### PR TITLE
fix: nil pointer dereference in CapacityReservationFromEC2 for Interruptible field

### LINKED ISSUE #9079: panic: nil pointer dereference in CapacityReservationFromEC2 — lo.Ternary eagerly evaluates *cr.Interruptible
### Description

**Observed Behavior:**

Karpenter panics with `runtime error: invalid memory address or nil pointer dereference` in `CapacityReservationFromEC2` when reconciling an EC2NodeClass that has `capacityReservationSelectorTerms` configured. The panic occurs every reconciliation cycle, making capacity reservation discovery non-functional.

```
panic: runtime error: invalid memory address or nil pointer dereference
github.com/aws/karpenter-provider-aws/pkg/apis/v1.CapacityReservationFromEC2({...}, 0x36ce83ffb348)
    github.com/aws/karpenter-provider-aws/pkg/apis/v1/ec2nodeclass_status.go:264 +0x2ea
```

**Expected Behavior:**

Karpenter should safely handle capacity reservations where the `Interruptible` field is nil (which is the default for standard ODCRs and capacity blocks).

**Root Cause:**

At [`ec2nodeclass_status.go:264`](https://github.com/aws/karpenter-provider-aws/blob/v1.10.0/pkg/apis/v1/ec2nodeclass_status.go#L264):

```go
Interruptible: lo.Ternary(cr.Interruptible == nil, false, *cr.Interruptible),
```

`lo.Ternary` is **not lazy** — Go evaluates all function arguments before the call. `*cr.Interruptible` is dereferenced unconditionally, panicking when the pointer is nil.

**Fix:**

Replace with `lo.FromPtrOr`:

```go
Interruptible: lo.FromPtrOr(cr.Interruptible, false),
```

We have validated this fix on a patched v1.10.0 build — capacity reservations (both capacity blocks and open ODCRs) are discovered successfully after the change.

**Reproduction Steps:**

1. Have any capacity reservation in the account where `Interruptible` is nil (this is the default for standard ODCRs and capacity blocks)
2. Configure an EC2NodeClass with `capacityReservationSelectorTerms` that matches the reservation
3. Observe continuous panic in the nodeclass controller

**Versions:**
- Karpenter: v1.10.0 (also present on `main`, v1.11.0, v1.11.1 — same code)
- Kubernetes: 1.31
- EKS

### PR DESCRIPTION
**Pull request by KeitaW**
## Description

Fixes #9079

`lo.Ternary` eagerly evaluates all arguments in Go, so:

```go
Interruptible: lo.Ternary(cr.Interruptible == nil, false, *cr.Interruptible),
```

dereferences `*cr.Interruptible` unconditionally, panicking when the pointer is nil. The `Interruptible` field may be absent from the EC2 API response depending on partition/region — when absent, the Go SDK deserializes the `*bool` as `nil`, triggering the panic.

Any user configuring `capacityReservationSelectorTerms` on an EC2NodeClass will hit this panic if any matched reservation has a nil `Interruptible` field.

## Changes

- Replace `lo.Ternary(cr.Interruptible == nil, false, *cr.Interruptible)` with `lo.FromPtrOr(cr.Interruptible, false)` which safely returns the default when nil
- Add test fixture with nil `Interruptible` and a regression test to `capacityreservation_test.go`

## How was this change tested?

- Added unit test with a capacity reservation fixture that omits the `Interruptible` field (nil pointer). The existing test fixtures all set `Interruptible: lo.ToPtr(false)`, which is why this bug was not caught.
- Validated on a live EKS cluster (Karpenter v1.10.0, us-west-2) with 6 capacity reservations (2 capacity blocks + 4 ODCRs), all with nil `Interruptible`. Before fix: continuous panic. After fix: all reservations discovered successfully.

### CODE REVIEW

File: pkg/controllers/nodeclass/capacityreservation_test.go [hunk 1]
```diff
@@ -131,9 +131,33 @@ var _ = Describe("NodeClass Capacity Reservation Reconciler", func() {
...
+					AvailabilityZone:       lo.ToPtr("test-zone-1a"),
+					InstanceType:           lo.ToPtr("m5.large"),
+					OwnerId:                lo.ToPtr(selfOwnerID),
+					InstanceMatchCriteria:  ec2types.InstanceMatchCriteriaTargeted,
+					CapacityReservationId:  lo.ToPtr("cr-nil-interruptible"),
+					AvailableInstanceCount: lo.ToPtr[int32](5),
+					State:                  ec2types.CapacityReservationStateActive,
+					ReservationType:        ec2types.CapacityReservationTypeDefault,
+					// Interruptible intentionally omitted (nil) — this is the default
+					// for standard ODCRs and capacity blocks returned by the EC2 API
```
Comment (ryan-mist) [inline pkg/controllers/nodeclass/capacityreservation_test.go:144]: This may actually partition dependent?

When I describe a standard ODCR I see interruptible as false:

```
aws ec2 describe-capacity-reservations --capacity-reservation-ids cr-017c7574f9fa6f97e

{
    "CapacityReservations": [
        {
            "CapacityReservationId": "cr-017c7574f9fa6f97e",
          ...
            "Interruptible": false
        }
    ]
}
```

Comment (ryan-mist) [reply pkg/controllers/nodeclass/capacityreservation_test.go:144]: Or its possible this field is not backfilled to existing reservations before IODCRs were introduced?

Comment (KeitaW) [reply pkg/controllers/nodeclass/capacityreservation_test.go:144]: Good point — this appears to be partition/region-dependent. In our account (us-west-2), the `Interruptible` field is completely absent from the API response for both capacity blocks and standard ODCRs, all created this week:

**Capacity block:**
```json
{
    "CapacityReservationId": "cr-0c4ef63e1c7be39bf",
    "InstanceType": "p5.48xlarge",
    "ReservationType": "capacity-block",
    "State": "active"
}
```

**Standard ODCR:**
```json
{
    "CapacityReservationId": "cr-00a3346b2c313c5be",
    "InstanceType": "g7e.12xlarge",
    "InstanceMatchCriteria": "open",
    "State": "active"
}
```

Neither includes `Interruptible` at all — the Go SDK deserializes the missing `*bool` as `nil`, triggering the panic. The fix handles both cases — field absent (nil) or explicitly set.

**Review by ryan-mist — APPROVED** [review]
LGTM 🚀

**Comment by KeitaW** [comment]
Thank you!
